.github/workflows/release.yml

```diff
@@ -0,0 +1,106 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  package_name: <YOUR PACKAGE NAME HERE>
+
+jobs:
+  version_check:
+    runs-on: ubuntu-latest
+    outputs:
+      v_package: ${{ steps.version.outputs.dbt_package_version}}
+      v_old: ${{ steps.previoustag.outputs.tag}}
+      new_version: ${{ steps.new_version_flag.outputs.new_version}}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+          fetch-depth: 0
+
+    - name: 'Get Previous tag'
+      id: previoustag
+      uses: "WyriHaximus/github-action-get-previous-tag@v1"
+      with:
+        fallback: 0.0.1 # Optional fallback tag to use when no tag can be found
+
+    - name: Get tag and package versions
+      id: version
+      run: |
+        echo "dbt_package_version=$(grep -oP "(?<=version: ')(.*)(?=')" dbt_project.yml)" >> $GITHUB_OUTPUT
+        echo "Project YAML version: $(grep -oP "(?<=version: ')(.*)(?=')" dbt_project.yml)"
+
+    - name: Fail if version already match
+      if: ${{ steps.previoustag.outputs.tag == steps.version.outputs.dbt_package_version }}
+      run: |
+        echo "Tag version (${{ steps.previoustag.outputs.tag }}) is the same as the version in project (${{ steps.version.outputs.dbt_package_version }}), no need to release."
```

Comment:
Does this fail it because we don't set the `new_version=yes`?

Reply:
Correct, I'll change the name though because I specifically made it _not_ fail given we might do small updates that aren't version changes.